### PR TITLE
Disable TrustedLaunch and StrictHostKeyChecking accept-new for 12sp5

### PR DESCRIPTION
**Pull request by mpagot**
Configure Standard security-type when creating VM using 12sp5 images.

Releted to [this article](https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch-faq?tabs=cli%2Cdebianbased#can-i-disable-trusted-launch-for-a-new-vm-deployment) 

- Related ticket: https://jira.suse.com/browse/TEAM-9714  https://jira.suse.com/browse/TEAM-9725

# Verification run:

sle-15-SP5-SapCloud-Azure-Payg-x86_64-BuildLATEST_AZURE_SLE15_5_PAYG-ipaddr2_azure_test
 -  http://openqaworker15.qa.suse.cz/tests/298897 :green_circle: 
 -  http://openqaworker15.qa.suse.cz/tests/298981 :green_circle: 

sle-12-SP5-SapCloud-Azure-Payg-x86_64-BuildLATEST_AZURE_SLE12_5_PAYG-ipaddr2_azure_test IPADDR2_TRUSTEDLAUNCH=0
-  http://openqaworker15.qa.suse.cz/tests/298895 :green_circle:  / :red_circle:  `az vm create` is fixed but one more problem pop-up related to ssh `StrictHostKeyChecking`

sle-12-SP5-SapCloud-Azure-Payg-x86_64-BuildLATEST_AZURE_SLE12_5_PAYG-ipaddr2_azure_test IPADDR2_TRUSTEDLAUNCH=0 IPADDR2_KEYCHECK_OLD=1
- http://openqaworker15.qa.suse.cz/tests/298979 , http://openqaworker15.qa.suse.cz/tests/298980  :green_circle:  / :red_circle:  ssh `StrictHostKeyChecking` is fixed but one more problem pop-up related to `crm cluster join`

### CODE REVIEW

File: t/22_ipaddr2.t [hunk 1]
```diff
@@ -255,6 +277,24 @@ subtest '[ipaddr2_internal_key_accept]' => sub {
...
+    $ipaddr2->redefine(assert_script_run => sub { push @calls, $_[0]; return; });
+    $ipaddr2->redefine(script_run => sub {
+            push @calls, $_[0];
+            if ($_[0] =~ /nc.*22/) { return 0; }
+            if ($_[0] =~ /ssh.*StrictHostKeyChecking/) { return 0; }
+            return 1; });
+    $ipaddr2->redefine(ipaddr2_bastion_pubip => sub { return '1.2.3.4'; });
+    $ipaddr2->redefine(ipaddr2_bastion_ssh_addr => sub { return 'AlessandroArtom@1.2.3.4'; });
+
+    my $ret = ipaddr2_internal_key_accept(key_checking => 'LuigiTorchi');
```
Comment (jankohoutek) [inline t/22_ipaddr2.t:292]: You already known what non confusing default values I prefer here, right? ;-)

Comment (mpagot) [reply t/22_ipaddr2.t:292]: yes, I think my reply in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20320#discussion_r1782243338 apply here too.

`ipaddr2_internal_key_accept(key_checking => 'LuigiTorchi');` , in conjunction with ...

File: lib/sles4sap/ipaddr2.pm [hunk 2]
```diff
@@ -119,6 +119,9 @@ Create a deployment in Azure designed for this specific test.
                       during the deployment so before any other part of the test can add
                       additional repo to test.
 
+=item B<trusted_launch> - Enable or disable Trusted Launch. Default 1: Enabled. If configured to 0 
+                          result in az vm create to be executed with '--security-type Standard'
```
Comment (lpalovsky) [inline lib/sles4sap/ipaddr2.pm:123]: I think the second part of conditional should be in future tense:
```
If configured to 0 the result in az vm create will be executed with '--security-type Standard'
```

And I think this should work as well:

```
If configured to 0 the result in az vm create is executed with '--security-type Standard'
```

File: tests/sles4sap/ipaddr2/deploy.pm [hunk 3]
```diff
@@ -39,7 +39,9 @@ sub run {
         diagnostic => get_var('IPADDR2_DIAGNOSTIC', 0),
         cloudinit => get_var('IPADDR2_CLOUDINIT', 1));
     $deployment{scc_code} = get_var('SCC_REGCODE_SLES4SAP') if (get_var('SCC_REGCODE_SLES4SAP'));
+    $deployment{trusted_launch} = 0 if (check_var('IPADDR2_TRUSTEDLAUNCH', 0));
```
Comment (lpalovsky) [inline tests/sles4sap/ipaddr2/deploy.pm:42]: Just curious. Is additional bracket around `check_var` required here? 
``` $deployment{trusted_launch} = 0 if (check_var('IPADDR2_TRUSTEDLAUNCH', 0));```

**Review by lpalovsky — APPROVED** [review]
LGTM

File: tests/sles4sap/ipaddr2/deploy.pm (hunk 3 above)
Comment (mpagot) [reply tests/sles4sap/ipaddr2/deploy.pm:42]: No are not. I like to always have `if ()` with braket.

File: tests/sles4sap/ipaddr2/configure.pm [hunk 4]
```diff
@@ -35,8 +35,13 @@ sub run {
     record_info("TEST STAGE", "Prepare all the ssh connections within the 2 internal VMs");
     my $bastion_ip = ipaddr2_bastion_pubip();
     ipaddr2_bastion_key_accept(bastion_ip => $bastion_ip);
-    ipaddr2_internal_key_accept(bastion_ip => $bastion_ip);
-    ipaddr2_internal_key_gen(bastion_ip => $bastion_ip);
+
+    my %int_key_args = (bastion_ip => $bastion_ip);
+    # unsupported option "accept-new" for default ssh used in 12sp5
+    $int_key_args{key_checking} = 'no' if (check_var('IPADDR2_KEYCHECK_OLD', '1'));
```
Comment (mpagot) [inline tests/sles4sap/ipaddr2/configure.pm:41]: `check_var('IPADDR2_KEYCHECK_OLD', '1')` is working
`check_var('IPADDR2_KEYCHECK_OLD', 1)` does not

File: t/22_ipaddr2.t (hunk 1 above)
Comment (alvarocarvajald) [reply t/22_ipaddr2.t:292]: > You already known what non confusing default values I prefer here, right? ;-)

This is not a default value. This is a unit test value.

File: tests/sles4sap/ipaddr2/configure.pm (hunk 4 above)
Comment (alvarocarvajald) [reply tests/sles4sap/ipaddr2/configure.pm:41]: Both should be the same: https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L744-L746

Do you have a link to where it didn't work. I'm curious about how it failed :)

**Pull request merged**